peer_client.go

```diff
@@ -84,80 +77,39 @@ type PeerConfig struct {
 	TraceGRPC bool
 }
 
-func NewPeerClient(conf PeerConfig) *PeerClient {
-	return &PeerClient{
+// NewPeerClient establishes a connection to a peer in a non-blocking fashion.
```

Comment:
it could be blocking if we added `grpc.WithBlock()`, but it's not recommended (see https://pkg.go.dev/google.golang.org/grpc#WithBlock)

Reply:
reverted to be non-blocking to preserve existing behavior.